### DISCUSSION TITLE
[RESOLVED] Regarding variable assignment

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#7689] [General]
I'm trying to understand the value assignment statement, looks like `i = 2` and `2 = i` both do the same thing? I expect `2 = i` to throw an error but it didn't

```
from int i
where i = 2
select i

from int i
where 2 = i
select i
```

**Comment by smowton** [comment]
These are equality constraints, not assignments, so they are symmetrical.
